### PR TITLE
Offline mode

### PR DESCRIPTION
**Pull request by dayan9265**
This PR creates the offline mode by:
- blocking creation button, upvote / downvote when offline
- use the offline cache of firestone (enabled automatically) to keep showing information when offline

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/cook4me/Cook4MeApp.kt [hunk 1]
```diff
@@ -41,6 +46,20 @@ fun Cook4MeApp(
         )
     }
 
+    val connectivityManager = LocalContext.current.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
+    val networkCallback = remember {
+        NetworkCallback()
+    }
+
```
Comment (JohnTomasTaylor) [inline app/src/main/java/ch/epfl/sdp/cook4me/Cook4MeApp.kt:53]: Nice! I would have forgotten that!

File: app/src/androidTest/java/ch/epfl/sdp/cook4me/ui/map/MapPermissionWrapper.kt [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
+// import androidx.compose.ui.test.junit4.createComposeRule
+// import androidx.compose.ui.test.onNodeWithText
+// import ch.epfl.sdp.cook4me.permissions.TestPermissionStatusProvider
+// import org.junit.Rule
+// import org.junit.Test
+
+// class TestMapPermissionWrapper {
+//    @get:Rule
+//    val composeTestRule = createComposeRule()
+
```
Comment (JohnTomasTaylor) [inline app/src/androidTest/java/ch/epfl/sdp/cook4me/ui/map/MapPermissionWrapper.kt:15]: Is it added to the backlog?

**Review by JohnTomasTaylor — APPROVED** [review]
Hey nice code! Have you added the test fixing as an issue to our backlog?

File: app/src/androidTest/java/ch/epfl/sdp/cook4me/ui/map/MapPermissionWrapper.kt (hunk 2 above)
Comment (dayan9265) [reply app/src/androidTest/java/ch/epfl/sdp/cook4me/ui/map/MapPermissionWrapper.kt:15]: No I forgot, I will add that now, thx for the reminder!

**Pull request merged**